pkg/operator/sync.go

```diff
@@ -617,6 +617,8 @@ func (optr *Operator) syncRequiredMachineConfigPools(_ *renderConfig) error {
 					isPoolStatusConditionTrue(pool, mcfgv1.MachineConfigPoolUpdated) {
 					continue
 				}
+				lastErr = fmt.Errorf("error required pool %s is not ready, retrying. Status: (total: %d, ready %d, updated: %d, unavailable: %d, degraded: %d)", pool.Name, pool.Status.MachineCount, pool.Status.ReadyMachineCount, pool.Status.UpdatedMachineCount, pool.Status.UnavailableMachineCount, pool.Status.DegradedMachineCount)
```

Comment:
Thanks for the fix! Just want to make sure I understand this correctly, the `pool.Status.ObservedGeneration` only gets updated to `pool.Generation` when the pool completes its update?

Reply:
The pool controller should bump `status.observedGeneration` whenever it sees a new `metadata.generation`.  The block above is accepting the pool if the `Updated=True` condition is set and the observed generation is new enough for us to know that that `Updated` is the controller's view vs. the current `spec`.  The restored error this thread is a comment on is what happens when the pool is not acceptable, because the pool controller hasn't noticed a `metadata.generation` bump, the pool isn't `Updated=True` yet, or both.